File: template-rest-assured-java/templates/src/test/java/{{package_name}}/support/utils/GetVariable.java
package {{package_name}}.support.utils;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

public class GetVariable {

    private static final String DATA_CONFIG = "src/test/resources/";
    private static Properties properties;

    public static String getProp(String value) {
        try {
            if (properties == null) {
                GetVariable.properties = new Properties();
                GetVariable.properties.load(new FileInputStream(DATA_CONFIG + getEnvironment() +".properties"));
            }

        } catch (IOException e) {
            e.printStackTrace();
        }

        return GetVariable.properties.getProperty(value);
    }

    public static String getEnvironment() {
        String environments = System.getProperty("environment");
        if (environments == null) {
            environments = "homolog";
        }
        return environments;
    }
}

File: template-rest-assured-kotlin/templates/src/test/kotlin/{{package_name}}/support/utils/GetVariable.kt
package {{package_name}}.support.utils

import java.util.Properties
import {{package_name}}.support.utils.GetVariable
import java.io.FileInputStream
import java.io.IOException

object GetVariable {
    private const val DATA_CONFIG = "src/test/resources/"
    private var properties: Properties? = null
    @JvmStatic
    fun getProp(value: String?): String {
        try {
            if (properties == null) {
                properties = Properties()
                properties!!.load(FileInputStream(DATA_CONFIG + environment + ".properties"))
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return properties!!.getProperty(value)
    }

    val environment: String
        get() {
            var environments = System.getProperty("environment")
            if (environments == null) {
                environments = "homolog"
            }
            return environments
        }
}